### PR TITLE
Replace solve with linsolve and some minor edits.

### PR DESCRIPTION
**Pull request by leosartaj**
@jcrist @flacjacket @aktech

### CODE REVIEW

File: sympy/series/formal.py [hunk 1]
```diff
@@ -589,19 +589,26 @@ def _transform_explike_DE(DE, g, x, order, syms):
         coeff = DE.coeff(Derivative(g(x), x, i))
         coeff = (coeff / highest_coeff).expand().collect(x)
         for t in Add.make_args(coeff):
-            if t.has(x):
```
Comment (aktech) [inline sympy/series/formal.py]: Why this was removed?

Comment (aktech) [inline sympy/series/formal.py]: Ok, I see that now.

File: sympy/series/formal.py [hunk 2]
```diff
@@ -589,19 +589,26 @@ def _transform_explike_DE(DE, g, x, order, syms):
...
+            break
+        temp.append(e)
+    else:
+        eq = temp
+    if eq:
+        sol = dict(zip(syms, (i for s in linsolve(eq, list(syms)) for i in s)))
+        if sol:
+            DE = DE.subs(sol)
+            DE = DE.factor().as_coeff_mul(Derivative)[1][0]
+            DE = DE.collect(Derivative(g(x)))
```
Comment (aktech) [inline sympy/series/formal.py]: The above three lines are untested now.

Comment (leosartaj) [inline sympy/series/formal.py]: This is used in the case of sin(x)*cos(x), which is in the test cases.

On 8 August 2015 at 01:11, AMiT Kumar <notifications@github.com> wrote:

> In sympy/series/formal.py
> <https://github.com/sympy/sympy/pull/9789#discussion_r36553895>:
>
> > -        DE = DE.factor().as_coeff_mul(Derivative)[1][0]
> > -        DE = DE.collect(Derivative(g(x)))
> > +            eq.append(t)
> > +    temp = []
> > +    for e in eq:
> > +        if e.has(x):
> > +            break
> > +        temp.append(e)
> > +    else:
> > +        eq = temp
> > +    if eq:
> > +        sol = dict(zip(syms, (i for s in linsolve(eq, list(syms)) for i in s)))
> > +        if sol:
> > +            DE = DE.subs(sol)
> > +            DE = DE.factor().as_coeff_mul(Derivative)[1][0]
> > +            DE = DE.collect(Derivative(g(x)))
>
> The above three lines are untested now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/sympy/pull/9789/files#r36553895>.
>



-- 
Regards
*Sartaj Singh*

*Mathematics and Computing*,
Indian Institute of Technology,
Varanasi - 221 005 INDIA

E-mail: singhsartaj94@gmail.com, *sartaj.singh.apm13@itbhu.ac.in
<sartaj.singh.apm13@itbhu.ac.in>*

Comment (aktech) [inline sympy/series/formal.py]: `sol` is never non-empty, i.e. `if sol:` is never True.

Comment (leosartaj) [inline sympy/series/formal.py]: Ah, nice catch. I guess you are right. I have fixed this now. Thanks!

**Comment by leosartaj** [comment]
Is this now ready to merge? @aktech @jcrist

File: sympy/series/formal.py (hunk 2 above)
Comment (jcrist) [inline sympy/series/formal.py]: Where is it fixed? Did you forget to push a commit?

Comment (leosartaj) [inline sympy/series/formal.py]: @c2d93b0fb99688b30cd6dfc04024e32b2753bb95 fixes it.

**Comment by leosartaj** [comment]
Thanks!

**Comment by aktech** [comment]
Oops, I almost forgot this PR, anyways Thanks @leosartaj for your work.

**Pull request merged**